Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Code/CMakeLists.txt

```diff
@@ -1,47 +1,103 @@
-#
+# 
 # Copyright (c) Contributors to the Open 3D Engine Project.
 # For complete copyright and license terms please see the LICENSE at the root of this distribution.
-#
+# 
 # SPDX-License-Identifier: Apache-2.0 OR MIT
-#
-#
+# 
+# 
+
+# Currently we are in the Code folder: ${CMAKE_CURRENT_LIST_DIR}
+# Get the platform specific folder ${pal_dir} for the current folder: ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME}
+# Note: o3de_pal_dir will take care of the details for us, as this may be a restricted platform
+#       in which case it will see if that platform is present here or in the restricted folder.
+#       i.e. It could here in our gem : Gems/DccScriptingInterface/Code/Platform/<platorm_name>  or
+#            <restricted_folder>/<platform_name>/Gems/DccScriptingInterface/Code
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
```

Comment:
Only the host platform names are present in the Code/Platform folder here. And it looks like the targets are editor targets. Is this an editor only module? If so, this needs to check the PAL_TRAIT_BUILD_HOST_TOOLS attribute before declaring this module at all,  or it will fail to build on any non-host clients like Android, iOS, etc

Reply:
@spham-amzn Our python tool template (that this gem was generated from) is expected to be an Editor only module as well, and does the check lower by wrapping the targets in a check for `PAL_TRAIT_BUILD_HOST_TOOLS`:

https://github.com/o3de/o3de/blob/development/Templates/PythonToolGem/Template/Code/CMakeLists.txt

Should I update the template to instead check for that earlier as you're suggesting and then bail out?

Reply:
In general, any editor targets should only be built when PAL_TRAIT_BUILD_HOST_TOOLS  is true

Reply:
It is already only declaring the targets inside an `if(PAL_TRAIT_BUILD_HOST_TOOLS)` check if that's what you mean. I linked the template file above. Does that look good?

Reply:
That does look good to me

Reply:
Cool thanks!

Reply:
@spham-amzn I think I fixed it right, can this comment be resolved?

Reply:
yes!